velox/exec/tests/HashJoinTest.cpp

```diff
@@ -206,6 +206,45 @@ TEST_F(HashJoinTest, filter) {
       "((t_k0 % 100) + (u_k0 % 100)) % 40 < 20");
 }
 
+TEST_F(HashJoinTest, joinSidesDifferentSchema) {
+  // In this join, the tables have different schema. The filter predicate uses
+  // a column from the right table before the left and the corresponding
+  // columns at the same channel number(1) have different types. This has been
+  // a source of errors in the join logic.
+
+  auto leftType = makeRowType({BIGINT(), VARCHAR(), BIGINT()}, "t_");
+  auto rightType = makeRowType({BIGINT(), REAL(), BIGINT()}, "u_");
+
+  auto leftBatch = std::dynamic_pointer_cast<RowVector>(
+      BatchMaker::createBatch(leftType, 100, *pool_));
+  auto rightBatch = std::dynamic_pointer_cast<RowVector>(
+      BatchMaker::createBatch(rightType, 100, *pool_));
+
+  std::string referenceQuery =
+      "SELECT t_k0 * t_k2/2 FROM "
```

Comment:
I'd expect "t_k0 * t_k2/2" projection after the join in the query plan, but I don't see it. I wonder how does this work? Can it be that the query returns empty resultset?

Reply:
Its possible that the query returns empty resultset. Without my fix, the query crashes when evaluating the predicate for existing rows. I generally focussed on getting the query to run without crashes.

For the purposes of this PR: I simplified the query so that a further projection isn't needed. Does this look better ?

Post this PR, I'm going to continue understand the join further so will add more cases.